gpytorch/likelihoods/gaussian_likelihood.py

```diff
@@ -2,42 +2,47 @@
 
 import math
 import warnings
+from typing import Any
+
+import torch
+from torch import Tensor
 
 from .. import settings
+from ..distributions import MultivariateNormal, base_distributions
 from ..likelihoods import Likelihood
-from ..distributions import base_distributions
-from .noise_models import HomoskedasticNoise
+from .noise_models import FixedGaussianNoise, HomoskedasticNoise, Noise
 
 
 class _GaussianLikelihoodBase(Likelihood):
-    """Base class for Gaussian Likelihoods, supporting general heteroskedastic noise models. """
+    """Base class for Gaussian Likelihoods, supporting general heteroskedastic noise models."""
+
+    def __init__(self, noise_covar: Noise, **kwargs: Any) -> None:
 
-    def __init__(self, noise_covar, **kwargs):
         super().__init__()
         param_transform = kwargs.get("param_transform")
         if param_transform is not None:
             warnings.warn("The 'param_transform' argument is now deprecated. If you want to use a different "
-                          "transformaton, specify a different 'noise_constraint' instead.")
+                          "transformaton, specify a different 'lengthscale_constraint' instead.")
```

Comment:
Why are we warning about lengthscale constraints in the likelihood? `noise_constraint` seems like it'd be correct.

Reply:
This is outdated issue. I pushed a fix to this to master to make it noise_constraint a couple of days ago. Should go away with a rebase